source/s3_client.c

```diff
@@ -422,7 +441,6 @@ static void s_s3_client_finish_destroy(void *user_data) {
 
     aws_retry_strategy_release(client->retry_strategy);
 
-    aws_event_loop_group_release(client->body_streaming_elg);
```

Comment:
This release now happens in the start_destroy and is now a condition for being able to get to finish_destroy.